src/gerbil/gxi

```diff
@@ -1,11 +1,48 @@
 #!/bin/sh
 set -eu
 
+
 if [ -z "${GERBIL_HOME:-}" ]; then
-    GERBIL_HOME=$(dirname $(cd ${0%/*} && echo $PWD))
+
+    # First, try whereis. We are looking for $GERBIL_HOME/lib/std, our stdlib
+    for gh in $(whereis -b gerbil |cut -d: -f2); do
+        if [ -d $gh/lib/std ]; then
+            GERBIL_HOME="$gh" ; break ;
+        fi;
+    done
+
+    # if that did not work, find out where we are and try that.
+    if [ -z "${GERBIL_HOME:-}" ]; then
+        GERBIL_BIN_LOCATION="$0"
+
+        # Check if we're a symlink
+        GERBIL_LINK=$(readlink $GERBIL_BIN_LOCATION || true)
```

Comment:
how reliable is this? We could skip the whereis test if this is reliable.

My concern is how long we are adding to the startup time with this dance.

Reply:
readlink isn't all that portable, and you need quotes around most every variable use including this one.